### PR TITLE
Minor improvements to the SearchBehavior

### PR DESCRIPTION
**Pull request by burzum**
* Made the config method configurable
* Added a convenience method to get the Manager object

### CODE REVIEW

File: src/Model/Behavior/SearchBehavior.php [hunk 1]
```diff
@@ -15,11 +16,13 @@ class SearchBehavior extends Behavior
      * @var array
      */
     protected $_defaultConfig = [
+        'searchConfigurationMethod' => 'searchConfiguration',
```
Comment (ADmad) [inline src/Model/Behavior/SearchBehavior.php]: The key it a bit too long, we can use `searchConfigMethod`. Also please add info about the config option in the docblock.

File: src/Model/Behavior/SearchBehavior.php [hunk 2]
```diff
@@ -54,7 +57,20 @@ public function findSearch(Query $query, array $options)
...
+        $valid = $this->_table->{$this->config('searchConfigurationMethod')}()->all();
         return ['search' => array_intersect_key($params, $valid)];
     }
+
+    /**
+     * Returns the search filter manager.
+     *
+     * @return \Search\Manager;
+     */
+    public function searchManager()
```
Comment (ADmad) [inline src/Model/Behavior/SearchBehavior.php]: Nice!

File: src/Model/Behavior/SearchBehavior.php (hunk 1 above)
Comment (burzum) [inline src/Model/Behavior/SearchBehavior.php]: Fixed.

**Comment by ADmad** [comment]
CS error :)

**Comment by burzum** [comment]
Ja ja ja... :unamused: 

Fixed.

**Comment by ADmad** [comment]
@burzum Can you please fix [these](https://scrutinizer-ci.com/g/FriendsOfCake/search/inspections/3cb738b5-4542-4c99-a874-65032017ae10/issues/files/src/Model/Behavior/SearchBehavior.php?status=new&orderField=path&order=asc) issues reported by scrutinizer.

**Pull request merged**